jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/ImageProvider.java

```diff
@@ -300,9 +300,10 @@ public void invalidateAll() {
 
 	private File findFileInPaths(String fileName) {
 		for (File path : this.lookupPaths) {
-			File searched = new File(path, fileName);
-			if (searched.isFile() && searched.canRead()) {
-				return searched;
+			for (File f : path.listFiles()) {
+				if (f.getName().equalsIgnoreCase(fileName) && f.isFile() && f.canRead()) {
```

Comment:
The condition of this should better be
`f.isFile() && f.canRead() && fileName.equalsIgnoreCase(f.getName()) `
This way, we can prevent a potential NullPointer if f.getName returns null and the simpler checks are done first.